File: backend/src/main/java/yeobaek/backend/club/dto/LastReadingResponse.java
package yeobaek.backend.club.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import java.time.LocalDateTime;

public record LastReadingResponse(
        @Schema(description = "모임 ID") Long clubId,
        @Schema(description = "모임 이름") String clubName,
        @Schema(description = "모임 도서") ClubBookResponse book,
        @Schema(description = "최근 열람 본문의 전체 순서") int lastReadPassageSequence,
        @Schema(description = "진도율 (0~100 정수, 반올림)") int progressRate,
        @Schema(description = "마지막으로 읽은 시간") LocalDateTime lastReadAt
) {
}

File: android/shared/src/commonMain/kotlin/com/yeobaek/data/dto/LastReadingResponse.kt
package com.yeobaek.data.dto

import com.yeobaek.data.model.LastReadingModel
import kotlinx.serialization.Serializable

@Serializable
data class LastReadingResponse(
    val book: Book,
    val clubId: Int,
    val clubName: String,
    val lastReadAt: String,
    val lastReadPassageSequence: Int,
    val progressRate: Int,
)

fun LastReadingResponse.toModel(): LastReadingModel = LastReadingModel(
    book = book.toModel(),
    clubId = clubId,
    clubName = clubName,
    lastReadAt = lastReadAt,
    lastReadPassageSequence = lastReadPassageSequence,
    progressRate = progressRate,
)
